vlc-android/src/org/videolan/vlc/gui/browser/NetworkBrowserFragment.java

```diff
@@ -170,15 +180,25 @@ public void onCtxAction(int position, int option) {
         final MediaWrapper mw = (MediaWrapper) (isRootDirectory() ? favoritesAdapter.getItem(position) : getAdapter().getItem(position));
         switch (option) {
             case Constants.CTX_NETWORK_ADD:
-                MediaDatabase.getInstance().addNetworkFavItem(mw.getUri(), mw.getTitle(), mw.getArtworkURL());
+                new BrowserFavRepository(VLCApplication.getAppContext()).addNetworkFavItem(mw.getUri(), mw.getTitle(), mw.getArtworkURL());
```

Comment:
You should have class variable for `BrowserFavRepository` instead of instantiating one for every operation